v_3_0_0/cloudconfig.go

```diff
@@ -36,6 +36,14 @@ func NewCloudConfig(config CloudConfigConfig) (*CloudConfig, error) {
 		return nil, microerror.Maskf(invalidConfigError, "config.Template must not be empty")
 	}
 
+	// Set default params.
+	if config.Params.MasterAPIDomain == "" {
+		config.Params.MasterAPIDomain = config.Params.Cluster.Kubernetes.API.Domain
+	}
+	if config.Params.Hyperkube.Apiserver.BindAddress == "" {
+		config.Params.Hyperkube.Apiserver.BindAddress = defaultHyperkubeApiserverBindAddress
+	}
```

Comment:
@calvix I'm also worried about backward compatibility. Those lines fallback to the old values when nothing is set.